packages/flutter_tools/templates/widget_preview_scaffold/lib/src/dtd/editor_service.dart.tmpl

```diff
@@ -41,14 +65,61 @@ mixin DtdEditorService {
           ).textDocument;
       }
     });
-    await dtd.streamListen(kEditorService);
+    await dtd.safeStreamListen(kEditorService);
+
+    dtd.onEvent(kServiceStream).listen((data) async {
+      switch (data) {
+        case DTDEvent(
+          kind: kServiceRegistered,
+          data: {
+            DtdParameters.service: kEditorService,
+            DtdParameters.method: kGetActiveLocation,
+          },
+        ):
+          // Manually retrieve the currently selected source file.
+          unawaited(_updateSelectedSourceFile());
+          _editorServiceAvailable.value = true;
+        case DTDEvent(
+          kind: kServiceRegistered,
+          data: {DtdParameters.service: kEditorService},
+        ):
+          _editorServiceAvailable.value = true;
```

Comment:
This additional check added in the last commit might affect already-shipped versions of Dart-Code (where people have enabled the preview), since those versions register many Editor services, but neither `getActiveLocation` or `navigateToCode` (they are only currently in pre-releases).

I don't know if that's a big deal though (it's an opt-in experiment, the update will ship next week, and you might have error handling that handles them not existing), but I'll let you decide :-)

Reply:
Yeah, I'm not too concerned here, especially with VSCode auto-updating the Dart-Code plugin. I don't think this would cause any noticeable breakages :)